core/history/history.go

```diff
@@ -148,6 +148,9 @@ func (h *history) Versions(id domain.FullID, lastVersionId string, limit int) (r
 		if e != nil {
 			return nil, e
 		}
+		if len(data) == 0 {
```

Comment:
How is it possible that data == 0? Because I guess we will always call func on first element